CandyMarket/CandyMarket/DataAccess/CandyRepo.cs

```diff
@@ -242,5 +242,47 @@ from UserCandies
                 return CandyThatHasBeenConsumed;
             }
         }
+
+        public UserCandy GetCandyForUser(int candyID, int userID)
+        {
+            var sql = @"select UserCandies.UserCandyId, UserCandies.DateReceived
+                        FROM Candies
+                        JOIN UserCandies ON Candies.CandyId = UserCandies.CandyId
+                        Where isConsumed = 0 
+                        AND Candies.CandyId = @candyID
+                        AND UserCandies.UserId = @userID
+                        ORDER BY UserCandies.DateReceived;";
+
+            using (var db = new SqlConnection(connectionString))
+            {
+                var parameters = new
+                {
+                    CandyId = candyID,
+                    UserId = userID
+                };
+
+                var candyToBeTraded = db.QueryFirstOrDefault<UserCandy>(sql, parameters);
+                return candyToBeTraded;
+            }
+        }
+        public bool UpdateTradedCandy(int receiverUserID, int userCandyIdToBeTraded)
+        {
+            // updates the usercandyid 
+            var sql = @"update UserCandies
+                        set UserId = @receiverUserID
+                        where UserCandyId = @candyToBeTraded
+                        ";
+
+            using (var db = new SqlConnection(connectionString))
+            {
+                var parameters = new 
+                {
+                    receiverUserID = receiverUserID,
```

Comment:
tiny change here.  If you look at line 280 in VS receiverUserID is kind of greyed out.  This is because it doesn't like the key and the value being the same exact name.  There are two options to fix this.  One is just to change the capitalization on the key (ReceiverUserID = receiverUserID), the other is that you could technically just list the variable name once without the =.  I don't recall the technical name for this, but basically you are telling it the key and value names are the same.  I think the capitalization change is a little more readable, but whichever you prefer is good!